### PR TITLE
[NTDLL] LdrpInitializeExecutionOptions(): Better handle failures

### PR DESCRIPTION
**Pull request by SergeGautherie**
And some related improvements.

### CODE REVIEW

File: dll/ntdll/ldr/ldrinit.c [hunk 1]
```diff
@@ -383,12 +386,13 @@ LdrQueryImageFileExecutionOptionsEx(IN PUNICODE_STRING SubKey,
...
-                                  OUT PVOID Buffer,
-                                  IN ULONG BufferSize,
-                                  OUT PULONG ReturnedLength OPTIONAL)
+LdrQueryImageFileExecutionOptions(
+    _In_ PUNICODE_STRING SubKey,
+    _In_ PCWSTR ValueName,
+    _In_ ULONG Type,
+    _Out_opt_ PVOID Buffer,
+    _In_ ULONG BufferSize,
+    _Out_opt_ PULONG ReturnedLength OPTIONAL)
```
Comment (HBelusca) [inline dll/ntdll/ldr/ldrinit.c:395]:
```suggestion
    _Out_opt_ PULONG ReturnedLength)
```

File: dll/ntdll/ldr/ldrinit.c [hunk 2]
```diff
@@ -1479,15 +1485,15 @@ LdrpInitializeExecutionOptions(PUNICODE_STRING ImagePathName, PPEB Peb, PHANDLE
...
+        Status = LdrQueryImageFileKeyOption(KeyHandle,
+                                            L"MinimumStackCommitInBytes",
+                                            REG_DWORD,
+                                            &MinimumStackCommit,
+                                            sizeof(MinimumStackCommit),
+                                            NULL);
 
         /* Update PEB's minimum stack commit if it's lower */
-        if (Peb->MinimumStackCommit < MinimumStackCommit)
+        if (NT_SUCCESS(Status) && Peb->MinimumStackCommit < MinimumStackCommit)
```
Comment (HBelusca) [inline dll/ntdll/ldr/ldrinit.c:1496]:
```suggestion
        if (NT_SUCCESS(Status) && (Peb->MinimumStackCommit < MinimumStackCommit))
```

File: sdk/include/ndk/umfuncs.h [hunk 3]
```diff
@@ -238,7 +238,7 @@ LdrQueryImageFileKeyOption(
     _In_ HANDLE KeyHandle,
     _In_ PCWSTR ValueName,
     _In_ ULONG Type,
-    _Out_ PVOID Buffer,
+    _Out_opt_ PVOID Buffer,
     _In_ ULONG BufferSize,
```
Comment (HBelusca) [inline sdk/include/ndk/umfuncs.h:242]:
```suggestion
    _In_opt_ ULONG BufferSize,
```

File: dll/ntdll/ldr/ldrinit.c [hunk 4]
```diff
@@ -181,12 +182,13 @@ LdrOpenImageFileOptionsKey(IN PUNICODE_STRING SubKey,
...
-                           IN ULONG Type,
-                           OUT PVOID Buffer,
-                           IN ULONG BufferSize,
-                           OUT PULONG ReturnedLength OPTIONAL)
+LdrQueryImageFileKeyOption(
+    _In_ HANDLE KeyHandle,
+    _In_ PCWSTR ValueName,
+    _In_ ULONG Type,
+    _Out_opt_ PVOID Buffer,
+    _In_ ULONG BufferSize,
```
Comment (HBelusca) [inline dll/ntdll/ldr/ldrinit.c:190]:
```suggestion
    _In_opt_ ULONG BufferSize,
```

File: sdk/include/ndk/umfuncs.h [hunk 5]
```diff
@@ -178,7 +178,7 @@ LdrQueryImageFileExecutionOptions(
     _In_ PUNICODE_STRING SubKey,
     _In_ PCWSTR ValueName,
     _In_ ULONG ValueSize,
-    _Out_ PVOID Buffer,
+    _Out_opt_ PVOID Buffer,
     _In_ ULONG BufferSize,
```
Comment (HBelusca) [inline sdk/include/ndk/umfuncs.h:182]:
```suggestion
    _In_opt_ ULONG BufferSize,
```

File: dll/ntdll/ldr/ldrinit.c [hunk 6]
```diff
@@ -345,13 +347,14 @@ LdrQueryImageFileKeyOption(IN HANDLE KeyHandle,
...
-                                    OUT PVOID Buffer,
-                                    IN ULONG BufferSize,
-                                    OUT PULONG ReturnedLength OPTIONAL,
-                                    IN BOOLEAN Wow64)
+LdrQueryImageFileExecutionOptionsEx(
+    _In_ PUNICODE_STRING SubKey,
+    _In_ PCWSTR ValueName,
+    _In_ ULONG Type,
+    _Out_opt_ PVOID Buffer,
+    _In_ ULONG BufferSize,
```
Comment (HBelusca) [inline dll/ntdll/ldr/ldrinit.c:355]:
```suggestion
    _In_opt_ ULONG BufferSize,
```

File: dll/ntdll/ldr/ldrinit.c [hunk 7]
```diff
@@ -383,12 +386,13 @@ LdrQueryImageFileExecutionOptionsEx(IN PUNICODE_STRING SubKey,
...
-                                  IN ULONG Type,
-                                  OUT PVOID Buffer,
-                                  IN ULONG BufferSize,
-                                  OUT PULONG ReturnedLength OPTIONAL)
+LdrQueryImageFileExecutionOptions(
+    _In_ PUNICODE_STRING SubKey,
+    _In_ PCWSTR ValueName,
+    _In_ ULONG Type,
+    _Out_opt_ PVOID Buffer,
+    _In_ ULONG BufferSize,
```
Comment (HBelusca) [inline dll/ntdll/ldr/ldrinit.c:394]:
```suggestion
    _In_opt_ ULONG BufferSize,
```

File: sdk/include/ndk/umfuncs.h (hunk 5 above)
Comment (Extravert-ir) [reply sdk/include/ndk/umfuncs.h:182]: `_Opt_` is useless for non-pointer params

File: dll/ntdll/ldr/ldrinit.c (hunk 1 above)
Comment (SergeGautherie) [reply dll/ntdll/ldr/ldrinit.c:395]: Sure.

File: dll/ntdll/ldr/ldrinit.c (hunk 2 above)
Comment (SergeGautherie) [reply dll/ntdll/ldr/ldrinit.c:1496]: Why? I would rather not.

Comment (tkreuzer) [reply dll/ntdll/ldr/ldrinit.c:1496]: This is part of our coding style. All sub-expressions need to be wrapped in braces for clarity.

File: dll/ntdll/ldr/ldrinit.c [hunk 8]
```diff
@@ -1811,7 +1816,7 @@ LdrpInitializeProcess(IN PCONTEXT Context,
     NtHeader = RtlImageNtHeader(Peb->ImageBaseAddress);
 
     /* Get the execution options */
-    Status = LdrpInitializeExecutionOptions(&ImagePathName, Peb, &OptionsKey);
+    (void)LdrpInitializeExecutionOptions(&ImagePathName, Peb, &OptionsKey);
```
Comment (tkreuzer) [inline dll/ntdll/ldr/ldrinit.c:1819]: Please use `if (!NT_SUCCESS(Status)) { DPRINT1("blabla faiure"); return Status; }` or `ASSERT(NT_SUCCESS(Status));`

File: dll/ntdll/ldr/ldrinit.c (hunk 2 above)
Comment (SergeGautherie) [reply dll/ntdll/ldr/ldrinit.c:1496]: > This is part of our coding style. All sub-expressions need to be wrapped in braces for clarity.

Is it? I see nothing about that on https://reactos.org/wiki/Coding_Style .
I would argue about unneeded parentheses being clearer or confusing.

File: dll/ntdll/ldr/ldrinit.c (hunk 8 above)
Comment (SergeGautherie) [reply dll/ntdll/ldr/ldrinit.c:1819]: Done.

**Pull request closed without merging**